src/wp-includes/blocks.php

```diff
@@ -361,12 +366,24 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	// Add `style` and `editor_style` for core blocks if missing.
 	if ( ! empty( $metadata['name'] ) && str_starts_with( $metadata['name'], 'core/' ) ) {
 		$block_name = str_replace( 'core/', '', $metadata['name'] );
-
+		$version    = ! $is_core_block && isset( $metadata['version'] ) ? $metadata['version'] : false;
 		if ( ! isset( $metadata['style'] ) ) {
 			$metadata['style'] = "wp-block-$block_name";
+			wp_register_style(
```

Comment:
I don't understand the benefit of registering these directly at this point rather than keeping responsibility for registering the styles in `register_block_style_handle`. Seems like it makes the logic more brittle if we don't have consistency in how block styles are registered.

Reply:
`register_block_style_handle` does a lot of things, including a `realpath` for a file that doesn't exist. See https://github.com/WordPress/wordpress-develop/blob/6eda6f0825a38f4bf1498abd24e492d0a80cfa0f/src/wp-includes/blocks.php#L225. This results in not hitting file system caches in PHP. This realpath lookup, what is we are trying avoid here. I had another idea with doing this. 

https://github.com/WordPress/wordpress-develop/commit/3eb647298a86ae3099059effb316458bc4cb0efc